### DISCUSSION TITLE
6.11 received a decent amount of RK3588 action recently

### DISCUSSION THREAD

**Discussion by RobMacGregor** [Joshua-Riek/ubuntu-rockchip#957] [General]
https://lore.kernel.org/lkml/172115794565.10577.13848078673989818961.pr-tracker-bot@kernel.org/T/

**Comment by Joshua-Riek** [comment]
I currently plan to release Ubuntu 24.10 with Linux 6.11. The only thing that may be a deal breaker for some is that HDMI is not working yet.

**Reply by wmport** [reply]
Sound output with some problems, it seems.

**Reply by coolva** [reply]
@Joshua-Riek,
Will 24.10/6.11 have ZFS support like the 6.1 kernel does?

**Reply by Joshua-Riek** [reply]
It should, though it may take me a little extra time to catch up on the kernel packaging changes. Unfortunately Ubuntu does not really document any of it yet it's essential to using DKMS modules such as ZFS.

I would also not expect an upgrade path from Ubuntu 22.04 / 24.04 to 24.10.

**Reply by coolva** [reply]
This is great to hear. THANK YOU for all your work!

**Reply by simcop2387** [reply]
Likely zfs will take a new release, 2.2.6 to get working 6.11 kernel support.  2.2.5 that was just released is the first one officially supporting 6.9 and unofficially 6.10 (it works everywhere but s390x arch).  That said there's no issues on their github tracking 6.11 compatibility but it hasn't been released yet so maybe it will just work with openzfs 2.2.5.

**Reply by Joshua-Riek** [reply]
Yeah, I had to disable zfs for now to get the kernel package building. But it's still a waiting game for 6.11 to release.

**Comment by RobMacGregor** [comment]
welp, that's definitely a deal breaker.

**Reply by Joshua-Riek** [reply]
To be honest I find it strange Collabora focused on the GPU before proper HDMI support.

**Reply by RobMacGregor** [reply]
Agreed, bit of a chicken or the egg situation.

**Reply by RobMacGregor** [reply]
https://gitlab.collabora.com/hardware-enablement/rockchip-3588/linux/-/issues/5

Here are the HDMI patches. Just waiting on mainline. It's getting close!

**Reply by RobMacGregor** [reply]
https://lore.kernel.org/lkml/2c97429e-4633-46f5-8e7a-ec6903901360@kernel.org/

TX HDMI patches submitted!

**Comment by stevepringle** [comment]
For those interested, the current state of progress with Collabora’s Rockchip RK3588 enablement project can be seen here:-mainline-status.md · main · hardware-enablement / Rockchip RK3588 upstream enablement efforts / Notes for Rockchip 3588 · GitLabgitlab.collabora.comRegards,Steve PringleSent from my iPadOn 29 Jul 2024, at 08:07, wmport ***@***.***> wrote:﻿
Sound output with some problems, it seems.

**Comment by Joshua-Riek** [comment]
I started to play with Ubuntu 24.10 and Linux 6.11 RC1 and so far it looks good. However I do not think HDMI will be fully supported and merged. I only plan to support a few boards, as I would like to primarily use these Ubuntu 24.10 images as a showcase of mainline Linux for RK3588. Additionally I will be using it heavily for my own development and staging my contributions to the Linux kernel.

**Reply by wmport** [reply]
nanopc-t6
test start SD 24.10 Next 4 [[dd16a76]](https://github.com/Joshua-Riek/ubuntu-rockchip/commit/dd16a76b8431fda60ec51a90564be1327950682f)
Hdmi does not work, display does not signal
boot indicator on the board works

**Reply by Joshua-Riek** [reply]
Hdmi has not been merged into mainline,  so expect it to not work. If it does work, it will have limited support and functionality.

**Reply by wmport** [reply]
ohhh ((

 28.016] (II) VESA: driver for VESA chipsets: vesa
[    28.016] (II) modesetting: Driver for Modesetting Kernel Drivers: kms
[    28.016] (II) FBDEV: driver for framebuffer: fbdev
[    28.016] (WW) xf86OpenConsole: setpgid failed: Operation not permitted
[    28.016] (WW) xf86OpenConsole: setsid failed: Operation not permitted
[    28.016] (WW) Falling back to old probe method for modesetting
[    28.016] (EE) open /dev/dri/card0: No such file or directory
[    28.016] (WW) Falling back to old probe method for fbdev
[    28.016] (II) Loading sub module "fbdevhw"
[    28.016] (II) LoadModule: "fbdevhw"
[    28.016] (II) Loading /usr/lib/xorg/modules/libfbdevhw.so
[    28.016] (II) Module fbdevhw: vendor="X.Org Foundation"
[    28.016] 	compiled for 1.21.1.11, module version = 0.0.2
[    28.016] 	ABI class: X.Org Video Driver, version 25.2
[    28.016] (EE) open /dev/fb0: No such file or directory
[    28.016] (WW) Falling back to old probe method for modesetting
[    28.016] (EE) open /dev/dri/card0: No such file or directory
[    28.016] (WW) Falling back to old probe method for fbdev
[    28.016] (II) Loading sub module "fbdevhw"
[    28.016] (II) LoadModule: "fbdevhw"
[    28.016] (II) Loading /usr/lib/xorg/modules/libfbdevhw.so
[    28.016] (II) Module fbdevhw: vendor="X.Org Foundation"
[    28.016] 	compiled for 1.21.1.11, module version = 0.0.2
[    28.016] 	ABI class: X.Org Video Driver, version 25.2
[    28.016] (EE) open /dev/fb0: No such file or directory
[    28.016] (EE) No devices detected.
[    28.016] (EE) 
Fatal server error:
[    28.016] (EE) no screens found(EE) 
[    28.017] (EE)

**Reply by HSTe** [reply]
https://lore.kernel.org/all/20240807-b4-rk3588-bridge-upstream-v3-0-60d6bab0dc7c@collabora.com/

**Comment by nrpetonr** [comment]
[50693e8](https://github.com/Joshua-Riek/ubuntu-rockchip/commit/50693e8acd8a3e7eabd386f97f6d6edfc4d9d989)  tried desktop version to see if there is any change on hdmi, not yet as joshua mentioned.  @Joshua-Riek  any chance to login to with desktop version over ssh, so i do not need to reflash to server version? ubuntu/ubuntu does not work...

**Reply by Joshua-Riek** [reply]
The desktop version does not have a pre installed user account. Hdmi is required. However, I've confirmed that HDMI does work. 

If it does not work for you try the other hdmi port and be sure to use a standard 1080p monitor.

**Reply by nrpetonr** [reply]
ups, have to look around for different monitor :-) meanwhile testing server version, seems to be good so far.

**Reply by wmport** [reply]
On an old samsung 22" hdmi-dvi port the panel lit up at 24.10-desktop! At 1080p resolution.
usb 2.0 no power, usb-c no power eitherь, but through an asix usb-c hub. Loved the picture of the invite )).
nanopc-t6

**Reply by nrpetonr** [reply]
is your system rendered with llvmpipe or with gpu? in my case, only llvmpipe. i assumed it should be gpu with panthor driver.

**Reply by Joshua-Riek** [reply]
Ubuntu has not updated their Linux firmware package for Ubuntu 24.10, which has the necessary firmware for the GPU driver.

Since Ubuntu 24.10 is not released, many packages have not been updated yet.

**Reply by HSTe** [reply]
Mesa also needs to be on version 24.1 to have support for panthor and mesa its on 24.0.9 at the moment in the repo

**Reply by wmport** [reply]
nanopc-ty6
After the initial boot in 1080p it started to boot in 2160p as well
Power supply on usb-c is unstable.

**Reply by Joshua-Riek** [reply]
> Mesa also needs to be on version 24.1 to have support for panthor and mesa its on 24.0.9 at the moment in the repo

Mesa was updated to 24.1 on 2024-08-06. I may need to enable the proposed apt repo.

https://launchpad.net/ubuntu/+source/mesa

**Reply by RobMacGregor** [reply]
Hot off the press! 
Mainline HDMI Bridge patch submitted a few days ago. We're almost there fellas! 

https://lore.kernel.org/linux-rockchip/20240914-b4-rk3588-bridge-upstream-v7-0-2b1348137123@collabora.com/

**Reply by JanV123** [reply]
Will 24.04.1 LTS also be getting 6.11 kernel? or is this exclusive to 24.10 only?

**Reply by ewaldc** [reply]
It is as easy as downloading the proper packages from `http://ppa.launchpad.net/jjriek/rockchip/ubuntu/pool/main/l/linux-rockchip/`, e.g. `*-6.11.0-1006-*`.
Afterwards:
```
apt install libcapstone4
dpkg -i linux-buildinfo-6.11.0-1006-rockchip_6.11.0-1006.6_arm64.deb linux-rockchip-headers-6.11.0-1006_6.11.0-1006.6_arm64.deb linux-headers-6.11.0-1006-rockchip_6.11.0-1006.6_arm64.deb linux-image-6.11.0-1006-rockchip_6.11.0-1006.6_arm64.deb linux-modules-6.11.0-1006-rockchip_6.11.0-1006.6_arm64.deb linux-modules-extra-6.11.0-1006-rockchip_6.11.0-1006.6_arm64.deb linux-rockchip-tools-6.11.0-1006_6.11.0-1006.6_arm64.deb linux-rockchip-tools-6.11.0-1006_6.11.0-1006.6_arm64.deb
```
If all went well, simply reboot and voila.  It is possible I got the order wrong of the packages (missing dependency), in which case you may have to reinstall the failed package.  If you want to boot back into your previous kernel, modify `/boot/extlinux/extlinux.conf` and change the `default` label (e.g. l1 to l2)

This is an easy way to see if 24.10 could work for you and e.g. the 6.11 kernel works on your HW.

Note: this will only work for ssh/console access as graphics/firmware versions in 24.04 won't be compatible with the 6.11 mainstream kernel.
